### ISSUE TITLE
PeerTube account email set to Mastodon username, why?

### ISSUE THREAD

**Issue by exstral** [agates/peertube-plugin-auth-oauth2-mastodon#3]
Why is the PeerTube account email set to the Mastodon username? That means that whenever PeerTube tries to send emails, it won't work.

Why cannot the Mastodon account email address be used instead?

**Comment by agates** [comment]
Mastodon's OAuth2 API doesn't return an email address, and peertube requires an email for registered users.  I don't like it either but I'm not sure what else to do about it :/

**Comment by agates** [comment]
And for the record, this is the main reason I forked the original oauth2 plugin.

**Comment by rasos** [comment]
Well, we've put a Keycloak instance in front of both, so that users get a single-sign-on experience. E-Mail as an attribute is handed over to both Mastodon and PeerTube.

**Comment by agates** [comment]
> Well, we've put a Keycloak instance in front of both, so that users get a single-sign-on experience. E-Mail as an attribute is handed over to both Mastodon and PeerTube.

@rasos  You should be able to use the standard -oauth plugin then, no?

**Comment by rasos** [comment]
@agates yes, we use the standard [auth-openid-connect 0.1.1](https://www.npmjs.com/package/peertube-plugin-auth-openid-connect) to authenticate with Keycloak. You probably could use any other IDM that has username & e-mail address as attributes.
